src/net.cpp

```diff
@@ -1166,11 +1166,23 @@ void static ProcessOneShot()
 void ThreadOpenConnections()
 {
     // Connect to specific addresses
-    if (mapArgs.count("-connect") && mapMultiArgs["-connect"].size() > 0)
+    if ((mapArgs.count("-connect-thinblock") && mapMultiArgs["-connect-thinblock"].size() > 0) ||
+        (mapArgs.count("-connect") && mapMultiArgs["-connect"].size() > 0))
+
     {
         for (int64_t nLoop = 0;; nLoop++)
         {
             ProcessOneShot();
+            BOOST_FOREACH(string strAddr, mapMultiArgs["-connect-thinblock"])
+            {
+                CAddress addr;
+                OpenNetworkConnection(addr, NULL, strAddr.c_str());
+                for (int i = 0; i < 10 && i < nLoop; i++)
+                {
+                    MilliSleep(500);
```

Comment:
this is used for making thinblock connection only when the config
setting of "connect-thinblock=<ip>" is used.     Is that what you're asking?

On 04/02/2016 6:35 PM, gandrewstone wrote:
>
> In src/net.cpp
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/8#discussion_r51970495>:
>
> >      {
> >          for (int64_t nLoop = 0;; nLoop++)
> >          {
> >              ProcessOneShot();
> > +            BOOST_FOREACH(string strAddr, mapMultiArgs["-connect-thinblock"])
> > +            {
> > +                CAddress addr;
> > +                OpenNetworkConnection(addr, NULL, strAddr.c_str());
> > +                for (int i = 0; i < 10 && i < nLoop; i++)
> > +                {
> > +                    MilliSleep(500);
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/8/files#r51970495>.
>